### DISCUSSION TITLE
Trying for autocomplete plugin

### DISCUSSION THREAD

**Discussion by ECon87** [saulpw/visidata#1038] [Q&A]
Hi,

As the title suggests I am trying to create an autocomplete feature for when I manually enter data in cells.The idea is similar to most GUI spreadsheet editors: take values from the current column, and use those as suggestions when I type new entries.

Below is my first attempt using the "prompt_toolkit" package to provide a completer. It works but it's not pretty.  The following are the main flaws of my solution:

1) It does not provide a nice drop-down menu as you type. Instead the prompt opens a new buffer I guess. I would prefer to figure out how to use the "editCell" command, if possible.

2) It requires extra keystrokes to go back to the original sheet ("enter" and "q"). However, the new cell value is recorded.

3) After returning to the original sheet, the status line is funky.

```
from visidata import Sheet, Column, BaseSheet
from prompt_toolkit import prompt
from prompt_toolkit.completion import WordCompleter


@Column.api
def get_values2(col, rows):
    values = [w for w in col.getValues(rows)]
    values = sorted(list(set(values)))
    values_completer = WordCompleter(values)
    choice = prompt('', completer=values_completer)
    return choice


Sheet.addCommand('1', 'get-values2',
                 'cursorCol.setValues([cursorRow], cursorCol.get_values2(rows))',
                 'get suggested values')

```

Any suggestions/feedback would be greatly appreciated.

Thanks.

**Comment by geekscrapy** [comment]
I am really interested in the completer, drop downs, and most importantly to me, syntax highlighting that prompt kit offers.
A while back I attempted to create a prompt for the main input prompt, as I wanted SQL completion and highlighting for one of my personal plugins. To say the least, I failed miserably so it's great to see someone else giving it a crack!

**Comment by saulpw** [comment]
Hi @ECon87, this is a really neat idea!  Unfortunately, it doesn't look like prompt_toolkit allows you to specify an existing curses screen to use, so you'd have to wrap the `prompt()` using `with SuspendCurses():`, which will lose the context of the alternate screen.  So you probably have to find another library to present it how you want.  I would love to see it in action though!

**Comment by frosencrantz** [comment]
@ECon87   Looking forward to this plugin. I hope you make it into a regular plugin and share with the world.
